experiments/kompanion/cmd/summary/summary.go

```diff
@@ -0,0 +1,398 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package summary
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/experiments/kompanion/pkg/utils"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	examples = `
+	# Summarize Config Connector resources across all namespaces, excludes \"kube\" namespaces by default
+	kompanion summary
+	`
+)
+
+func BuildSummaryCmd() *cobra.Command {
+	opts := NewSummaryOptions()
+	cmd := &cobra.Command{
+		Use:     "summary",
+		Short:   "summarize Config Connector resources",
+		Example: examples,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return RunSummarize(cmd.Context(), opts)
+		},
+		Args: cobra.ExactArgs(0),
+	}
+
+	flags := cmd.Flags()
+	opts.AddFlags(flags)
+
+	return cmd
+}
+
+type Result struct {
+	// Map[namespace, Map[gk, Map[status, count]]]
+	counts map[string]map[string]map[string]int
+	// Map[gk, Map[status, count]]
+	totals map[string]map[string]int
+	// secures access to counts & totals
+	countLock sync.Mutex
+}
+
+func (r *Result) increment(namespace string, gk string, status string) {
+	r.countLock.Lock()
+	defer r.countLock.Unlock()
+	if r.counts == nil {
+		r.counts = make(map[string]map[string]map[string]int)
+	}
+	nsMap, present := r.counts[namespace]
+	if !present {
+		nsMap = make(map[string]map[string]int)
+		r.counts[namespace] = nsMap
+	}
+	gkMap, present := nsMap[gk]
+	if !present {
+		gkMap = make(map[string]int)
+		nsMap[gk] = gkMap
+	}
+	count := gkMap[status]
+	// Can ignore not present because count will default to 0.
+	gkMap[status] = count + 1
+
+	if r.totals == nil {
+		r.totals = make(map[string]map[string]int)
+	}
+	gkMap, present = r.totals[gk]
+	if !present {
+		gkMap = make(map[string]int)
+		r.totals[gk] = gkMap
+	}
+	count = gkMap[status]
+	// Can ignore not present because count will default to 0.
+	gkMap[status] = count + 1
+}
+
+func (r *Result) Print() {
```

Comment:
I think this format is fine but it may be nice to support tabulation, json or other formats in the future.

Reply:
+1.  Even just a CSV format. See if its popular and if it is how users would like to consumer the results.